core/commands/cat.go

```diff
@@ -128,8 +128,14 @@ func cat(ctx context.Context, api iface.CoreAPI, paths []string, offset int64, m
 			return nil, 0, err
 		}
 
-		file, ok := f.(files.File)
-		if !ok {
+		var file files.File
+		switch f := f.(type) {
+		case files.File:
+			file = f
+		case files.Directory:
+			return nil, 0, iface.ErrIsDir
+		default:
+			// TODO: Consider returning "error not supported" for symlinks.
```

Comment:
Doesn't go-unixfs do that? https://github.com/ipfs/go-unixfs/blob/master/io/dagreader.go#L80

Reply:
Yeah, I just didn't feel like adding that error to the core API yet (I'm being lazy and trying to cut a release...).

Reply:
@magik6k https://github.com/ipfs/interface-go-ipfs-core/pull/5